test/util/discovery/discovery.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	"github.com/openshift/sriov-network-operator/test/util/client"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Enabled indicates whether test discovery mode is enabled.
+func Enabled() bool {
+	discoveryMode, _ := strconv.ParseBool(os.Getenv("DISCOVERY_MODE"))
+	return discoveryMode
+}
+
+func getSriovNodes(clients *client.ClientSet, sriovNodes []string) ([]corev1.Node, error) {
```

Comment:
How about just iterating sriovNodes and Get-ing each node?

Reply:
Done,